python3Packages.joblib fails on Darwin

**Describe the bug**

`nix-build -A python3Packages.joblib` fails on Darwin.

**To Reproduce**
Steps to reproduce the behavior:
1. `nix-build -A python3Packages.joblib`

**Expected behavior**
Package builds

**Additional context**

Error log:

<details>
<summary>Error log</summary>

```console
running install tests
============================= test session starts ==============================
platform darwin -- Python 3.7.4, pytest-4.6.3, py-1.7.0, pluggy-0.12.0
rootdir: /private/var/folders/ck/ht80_l5d4c9_tnwmd5lqsxj40000gn/T/nix-build-python3.7-joblib-0.13.2.drv-0/joblib-0.13.2, inifile: setup.cfg, testpaths: joblib
collected 1109 items / 1 deselected / 2 skipped / 1106 selected

joblib/__init__.py s                                                     [  0%]
joblib/parallel.py ss                                                    [  0%]
joblib/test/test_backports.py s....                                      [  0%]
joblib/test/test_disk.py ........                                        [  1%]
joblib/test/test_format_stack.py s...                                    [  1%]
joblib/test/test_func_inspect.py s....................................   [  5%]
joblib/test/test_hashing.py sssssssssss................................. [  9%]
........................................................................ [ 15%]
........................................................................ [ 22%]
........................................................................ [ 28%]
........................................................................ [ 35%]
........................................................................ [ 41%]
........................................................................ [ 48%]
....................ss.............                                      [ 51%]
joblib/test/test_init.py .                                               [ 51%]
joblib/test/test_logger.py .                                             [ 51%]
joblib/test/test_memmapping.py ssssssssssssss..                          [ 52%]
joblib/test/test_memory.py ss........................................... [ 56%]
......                                                                   [ 57%]
joblib/test/test_module.py ..                                            [ 57%]
joblib/test/test_my_exceptions.py ...                                    [ 57%]
joblib/test/test_numpy_pickle.py sssssssssssssssssssss.................. [ 61%]
...............................................................s         [ 67%]
joblib/test/test_numpy_pickle_compat.py .                                [ 67%]
joblib/test/test_numpy_pickle_utils.py ..                                [ 67%]
joblib/test/test_parallel.py sssss...................................... [ 71%]
........................................................................ [ 77%]
........................................................................ [ 84%]
........FFF.......[DEBUG:MainProcess:Thread-505] worker handler exiting
[DEBUG:MainProcess:Thread-508] worker handler exiting
[DEBUG:MainProcess:Thread-506] task handler got sentinel
[DEBUG:MainProcess:Thread-506] task handler sending sentinel to result handler
[DEBUG:MainProcess:Thread-506] task handler sending sentinel to workers
[DEBUG:MainProcess:Thread-506] task handler exiting
[DEBUG:MainProcess:Thread-486] joining task handler
[DEBUG:MainProcess:Thread-487] joining task handler
...................................................... [ 90%]
....s.ss.s...s.ss.s...s.ss.s............................................ [ 97%]
...............ss...                                                     [ 99%]
joblib/test/test_store_backends.py ...                                   [ 99%]
joblib/test/test_testing.py .....                                        [ 99%]
joblib/test/data/create_numpy_pickle.py s                                [100%]

=================================== FAILURES ===================================
________________ test_dispatch_multiprocessing[multiprocessing] ________________

backend = 'multiprocessing'

    @with_multiprocessing
    @parametrize('backend', PARALLEL_BACKENDS)
    def test_dispatch_multiprocessing(backend):
        """ Check that using pre_dispatch Parallel does indeed dispatch items
            lazily.
        """
>       manager = mp.Manager()

joblib/test/test_parallel.py:448:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/context.py:56: in Manager
    m.start()
/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/managers.py:567: in start
    self._address = reader.recv()
/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/connection.py:250: in recv
    buf = self._recv_bytes()
/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/connection.py:407: in _recv_bytes
    buf = self._recv(4)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <multiprocessing.connection.Connection object at 0x10adf0e50>, size = 4
read = <built-in function read>

    def _recv(self, size, read=_read):
        buf = io.BytesIO()
        handle = self._handle
        remaining = size
        while remaining > 0:
            chunk = read(handle, remaining)
            n = len(chunk)
            if n == 0:
                if remaining == size:
>                   raise EOFError
E                   EOFError

/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/connection.py:383: EOFError
----------------------------- Captured stderr call -----------------------------
[INFO:SyncManager-338:MainThread] child process calling self.run()
[INFO:SyncManager-338:MainThread] created temp directory /private/var/folders/ck/ht80_l5d4c9_tnwmd5lqsxj40000gn/T/nix-build-python3.7-joblib-0.13.2.drv-0/pymp-grdx4c0r
[INFO:SyncManager-338:MainThread] process shutting down
[DEBUG:SyncManager-338:MainThread] running all "atexit" finalizers with priority >= 0
[DEBUG:SyncManager-338:MainThread] running the remaining "atexit" finalizers
Process SyncManager-338:
Traceback (most recent call last):
  File "/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/process.py", line 297, in _bootstrap
    self.run()
  File "/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/process.py", line 99, in run
    self._target(*self._args, **self._kwargs)
  File "/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/managers.py", line 589, in _run_server
    server = cls._Server(registry, address, authkey, serializer)
  File "/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/managers.py", line 145, in __init__
    self.listener = Listener(address=address, backlog=16)
  File "/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/connection.py", line 438, in __init__
    self._listener = SocketListener(address, family, backlog)
  File "/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/connection.py", line 581, in __init__
    self._socket.bind(address)
OSError: AF_UNIX path too long
[INFO:SyncManager-338:MainThread] process exiting with exitcode 1
_____________________ test_dispatch_multiprocessing[loky] ______________________

backend = 'loky'

    @with_multiprocessing
    @parametrize('backend', PARALLEL_BACKENDS)
    def test_dispatch_multiprocessing(backend):
        """ Check that using pre_dispatch Parallel does indeed dispatch items
            lazily.
        """
>       manager = mp.Manager()

joblib/test/test_parallel.py:448:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/context.py:56: in Manager
    m.start()
/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/managers.py:567: in start
    self._address = reader.recv()
/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/connection.py:250: in recv
    buf = self._recv_bytes()
/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/connection.py:407: in _recv_bytes
    buf = self._recv(4)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <multiprocessing.connection.Connection object at 0x109ef9e50>, size = 4
read = <built-in function read>

    def _recv(self, size, read=_read):
        buf = io.BytesIO()
        handle = self._handle
        remaining = size
        while remaining > 0:
            chunk = read(handle, remaining)
            n = len(chunk)
            if n == 0:
                if remaining == size:
>                   raise EOFError
E                   EOFError

/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/connection.py:383: EOFError
----------------------------- Captured stderr call -----------------------------
[INFO:SyncManager-339:MainThread] child process calling self.run()
[INFO:SyncManager-339:MainThread] created temp directory /private/var/folders/ck/ht80_l5d4c9_tnwmd5lqsxj40000gn/T/nix-build-python3.7-joblib-0.13.2.drv-0/pymp-e04hzocl
[INFO:SyncManager-339:MainThread] process shutting down
[DEBUG:SyncManager-339:MainThread] running all "atexit" finalizers with priority >= 0
[DEBUG:SyncManager-339:MainThread] running the remaining "atexit" finalizers
Process SyncManager-339:
Traceback (most recent call last):
  File "/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/process.py", line 297, in _bootstrap
    self.run()
  File "/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/process.py", line 99, in run
    self._target(*self._args, **self._kwargs)
  File "/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/managers.py", line 589, in _run_server
    server = cls._Server(registry, address, authkey, serializer)
  File "/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/managers.py", line 145, in __init__
    self.listener = Listener(address=address, backlog=16)
  File "/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/connection.py", line 438, in __init__
    self._listener = SocketListener(address, family, backlog)
  File "/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/connection.py", line 581, in __init__
    self._socket.bind(address)
OSError: AF_UNIX path too long
[INFO:SyncManager-339:MainThread] process exiting with exitcode 1
___________________ test_dispatch_multiprocessing[threading] ___________________

backend = 'threading'

    @with_multiprocessing
    @parametrize('backend', PARALLEL_BACKENDS)
    def test_dispatch_multiprocessing(backend):
        """ Check that using pre_dispatch Parallel does indeed dispatch items
            lazily.
        """
>       manager = mp.Manager()

joblib/test/test_parallel.py:448:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/context.py:56: in Manager
    m.start()
/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/managers.py:567: in start
    self._address = reader.recv()
/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/connection.py:250: in recv
    buf = self._recv_bytes()
/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/connection.py:407: in _recv_bytes
    buf = self._recv(4)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <multiprocessing.connection.Connection object at 0x10ad69bd0>, size = 4
read = <built-in function read>

    def _recv(self, size, read=_read):
        buf = io.BytesIO()
        handle = self._handle
        remaining = size
        while remaining > 0:
            chunk = read(handle, remaining)
            n = len(chunk)
            if n == 0:
                if remaining == size:
>                   raise EOFError
E                   EOFError

/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/connection.py:383: EOFError
----------------------------- Captured stderr call -----------------------------
[INFO:SyncManager-340:MainThread] child process calling self.run()
[INFO:SyncManager-340:MainThread] created temp directory /private/var/folders/ck/ht80_l5d4c9_tnwmd5lqsxj40000gn/T/nix-build-python3.7-joblib-0.13.2.drv-0/pymp-c4ghaxyf
[INFO:SyncManager-340:MainThread] process shutting down
[DEBUG:SyncManager-340:MainThread] running all "atexit" finalizers with priority >= 0
[DEBUG:SyncManager-340:MainThread] running the remaining "atexit" finalizers
Process SyncManager-340:
Traceback (most recent call last):
  File "/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/process.py", line 297, in _bootstrap
    self.run()
  File "/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/process.py", line 99, in run
    self._target(*self._args, **self._kwargs)
  File "/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/managers.py", line 589, in _run_server
    server = cls._Server(registry, address, authkey, serializer)
  File "/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/managers.py", line 145, in __init__
    self.listener = Listener(address=address, backlog=16)
  File "/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/connection.py", line 438, in __init__
    self._listener = SocketListener(address, family, backlog)
  File "/nix/store/pwqcjzvh0slsrx1qylc9b95k216bd6zs-python3-3.7.4/lib/python3.7/multiprocessing/connection.py", line 581, in __init__
    self._socket.bind(address)
OSError: AF_UNIX path too long
[INFO:SyncManager-340:MainThread] process exiting with exitcode 1
======= 3 failed, 1028 passed, 79 skipped, 1 deselected in 81.40 seconds =======
[INFO:MainProcess:MainThread] process shutting down
[DEBUG:MainProcess:MainThread] running all "atexit" finalizers with priority >= 0
[DEBUG:MainProcess:MainThread] Interpreter shutting down. Waking up queue_manager_threads [(<Thread(QueueManagerThread, started daemon 123145379618816)>, <joblib.externals.loky.process_executor._ThreadWakeup object at 0x10ad33190>)]
[DEBUG:MainProcess:QueueManagerThread] queue management thread shutting down
[DEBUG:MainProcess:QueueManagerThread] closing call_queue
[DEBUG:MainProcess:QueueManagerThread] telling queue thread to quit
[DEBUG:MainProcess:QueueManagerThread] joining processes
[DEBUG:MainProcess:QueueFeederThread] feeder thread got sentinel -- exiting
[DEBUG:MainProcess:QueueManagerThread] queue management thread clean shutdown of worker processes: []
[DEBUG:MainProcess:MainThread] running the remaining "atexit" finalizers
builder for '/nix/store/42791m8ybfxma4cz058haj5criwiwlvk-python3.7-joblib-0.13.2.drv' failed with exit code 1
error: build of '/nix/store/42791m8ybfxma4cz058haj5criwiwlvk-python3.7-joblib-0.13.2.drv' failed
```

</details>

**Metadata**
Please run `nix run nixpkgs.nix-info -c nix-info -m` and paste the result.

 - host os: `Darwin 18.6.0, macOS 10.14.5`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.2.2`
 - channels(yl): `"darwin, nixpkgs-19.09pre185256.5b8b5a694c7"`
 - nixpkgs: `/run/current-system/nixpkgs`


Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: python3Packages.joblib
```

Possibly related to #21437

--- a/pkgs/development/python-modules/joblib/default.nix
+++ b/pkgs/development/python-modules/joblib/default.nix
@@ -2,6 +2,7 @@
 , buildPythonPackage
 , fetchPypi
 , fetchpatch
+, stdenv
 , sphinx
 , numpydoc
 , pytest
@@ -38,10 +39,10 @@ buildPythonPackage rec {
   checkInputs = [ sphinx numpydoc pytest ];
   propagatedBuildInputs = [ python-lz4 ];
 
-  # test_disk_used is broken
-  # https://github.com/joblib/joblib/issues/57
+  # test_disk_used is broken: https://github.com/joblib/joblib/issues/57
+  # test_dispatch_multiprocessing is broken only on Darwin.
   checkPhase = ''
-    py.test joblib -k "not test_disk_used"
+    py.test -k 'not test_disk_used${lib.optionalString (stdenv.isDarwin) " and not test_dispatch_multiprocessing"}' joblib/test
   '';
 
   meta = {

